lib/connectors/memory.js

```diff
@@ -220,8 +220,12 @@ Memory.prototype._createSync = function(model, data, fn) {
     this.collection(model, {});
   }
 
-  if (this.collection(model)[id])
-    return fn(new Error(g.f('Duplicate entry for %s.%s', model, idName)));
+  if (this.collection(model)[id]) {
+    var errMsg = g.f('Duplicate entry for %s.%s', model, idName);
+    var error = new Error(errMsg);
```

Comment:
Nitpick - feel free to ignore: I would personally simplify this lines to a single statement.

```js
var error = new Error(g.f(
  'Duplicate entry for %s.%s', 
  model, idName));
```